lib/new_relic/agent/instrumentation/roda.rb

```diff
@@ -0,0 +1,40 @@
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+require_relative 'roda/instrumentation'
+require_relative 'roda/chain'
+require_relative 'roda/prepend'
+
+DependencyDetection.defer do
+  named :roda
+
+  depends_on do
+    defined?(Roda) &&
+      Roda::RodaVersion >= '3.19.0' &&
```

Comment:
`Roda::RodaVersion` is a simple string, and you won't want to compare 2 strings for their numeric content.

I believe you are attempting to support Roda v3.19.0 and above, but currently this check succeeds for Roda v3.2.0 as well given that `2` >= `1`. Using a `Gem::Version` instance for comparison should yield your desired behavior.

```suggestion
      Gem::Version.new(Roda::RodaVersion) >= '3.19.0' &&
```

Reply:
Thank you—we've definitely been over this one! [All fixed](https://github.com/newrelic/newrelic-ruby-agent/pull/2144/files#diff-7f80c12f838c8bc306fce6b66036c7760f13728f57ce6622f1de94a966c40c6aR14).